k4FWCore/scripts/k4run

```diff
@@ -6,6 +6,7 @@ import argparse
 import logging
 
 from k4FWCore.utils import load_file
+from ROOT import gROOT
```

Comment:
Ugh, this will make calling `k4run` slower, in particular the first time because of the ROOT imports:
```
{jcarcell@lxplus969.cern.ch} ~ % time python -c "from ROOT import gROOT"
python -c "from ROOT import gROOT"  0.59s user 0.36s system 8% cpu 11.352 total
{jcarcell@lxplus969.cern.ch} ~ % time python -c "from ROOT import gROOT"
python -c "from ROOT import gROOT"  0.54s user 0.30s system 65% cpu 1.300 total
```

Reply:
I know :sunglasses:

Reply:
Can we move the import to after the `dry_run` check. Or alternatively, hide this behind a dedicated flag for `k4run`?

Alternatively, which algorithm is causing this, and can't that be fixed to not create a Canvas?

Reply:
> Can we move the import to after the `dry_run` check

yes

> Alternatively, which algorithm is causing this, and can't that be fixed to not create a Canvas?

It is https://github.com/iLCSoft/CLICPerformance/blob/master/Tracking/src/TrackChecker.cc
but the creation of the canvas is intended :( 

We could also turn on batch mode there but without explicitly searching I would think it will not be the only existing processors, just the only one we use regularly...

In the end I don't care were and how we fix this, but I still think having ROOT in batch mode is the most future proof solution. Who knows when they make something else suddenly produce graphics output :)

Reply:
OK. In that case I would hide it behind a flag for `k4run` (that disables batch mode).

Reply:
I added a flag that disables batch mode now but given the previous discussion isn't that the opposite of what you wanted? :D

Reply:
Ah yes, sorry. I actually wanted the flag the other way around. Opt-in to non-batch mode (too many negatives flying around here).

Reply:
But this is opt-in to non-batch mode?

Reply:
The other way around I am not interested in as we will always have to turn it on and then I can just keep it in CLDConfig :shrug: so then we can also just close this :)

Reply:
> But this is opt-in to non-batch mode?

You are right. And I think this is what we want. With this flag, the default is to run in batch mode, but it gives users the possibility to turn it off and get outputs (if anything creates them).

One issue with this is that algorithms are in principle free to switch things again in their implementation, but I suppose there we simply have to tell people to not do that.